### PR TITLE
[13.0][MIG]  account_banking_mandate

### PR DESCRIPTION
**Pull request by misern2**
Standard migration

Cherry pick included #678

### CODE REVIEW

**Comment by sergio-teruel** [comment]
Can you add https://github.com/OCA/bank-payment/pull/678?

**Comment by misern2** [comment]
@sergio-teruel Cherry pick added.

**Comment by sergio-teruel** [comment]
Thanks @misern2 , Can you check travis?? I know that this module depends on others but there is other travis fails..

**Comment by sergio-teruel** [comment]
@misern2 Can you do a rebase to fix isort conflicts??

**Comment by misern2** [comment]
@sergio-teruel Rebase OK. Waiting for pending dependences.

File: account_banking_mandate/README.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+This module is part of the `OCA/bank-payment <https://github.com/OCA/bank-payment/tree/13.0/account_banking_mandate>`_ project on GitHub.
+
+You are welcome to contribute. To learn how please visit https://odoo-community.org/page/Contribute.
```
Comment (sergio-teruel) [inline account_banking_mandate/README.rst:102]: You must apply oca-gen-readme script

File: account_banking_mandate/migrations/12.0.2.0.0/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from openupgradelib import openupgrade, openupgrade_90
+
+
+@openupgrade.migrate(use_env=True)
+def migrate(env, version):
+    column = openupgrade.get_legacy_name("scan")
+    if openupgrade.column_exists(env.cr, "account_banking_mandate", column):
+        openupgrade_90.convert_binary_field_to_attachment(
+            env, {"account.banking.mandate": [("scan", None)]}
+        )
```
Comment (sergio-teruel) [inline account_banking_mandate/migrations/12.0.2.0.0/post-migration.py]: Remove this file

File: account_banking_mandate/migrations/12.0.2.0.0/pre-migration.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+from openupgradelib import openupgrade
+
+column_renames = {"account_banking_mandate": [("scan", None)]}
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if openupgrade.column_exists(env.cr, "account_banking_mandate", "scan"):
+        openupgrade.rename_columns(env.cr, column_renames)
```
Comment (sergio-teruel) [inline account_banking_mandate/migrations/12.0.2.0.0/pre-migration.py]: Remove this file

File: account_banking_mandate/models/account_banking_mandate.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+        related="partner_bank_id.partner_id",
+        string="Partner",
+        store=True,
+        index=True,
+    )
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        string="Company",
+        required=True,
+        default=lambda self: self.env["res.company"]._company_default_get(
```
Comment (sergio-teruel) [inline account_banking_mandate/models/account_banking_mandate.py]: in v13 company is in env
https://github.com/odoo/odoo/blob/76a6d0475872f9ffa21cb917f13bc553c720eafd/addons/sale/models/sale.py#L226

File: account_banking_mandate/models/account_banking_mandate.py [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+            "account.banking.mandate"
+        ),
+    )
+    unique_mandate_reference = fields.Char(
+        string="Unique Mandate Reference", track_visibility="onchange", copy=False
+    )
+    signature_date = fields.Date(
+        string="Date of Signature of the Mandate", track_visibility="onchange"
+    )
+    scan = fields.Binary(string="Scan of the Mandate", attachment=True)
```
Comment (sergio-teruel) [inline account_banking_mandate/models/account_banking_mandate.py]: attachment=True is the default value... so you can remove it

**Comment by pedrobaeza** [comment]
Please include #678

**Comment by misern2** [comment]
#678 already included

**Comment by pedrobaeza** [comment]
Great, thanks, I didn't see it. Let's wait for the dependency to be merged.

**Comment by HaraldPanten** [comment]
Hi @misern2 ,

Could you check failures?

https://github.com/OCA/bank-payment/pull/651 has been merged.

Thanks!

**Comment by misern2** [comment]
@HaraldPanten  done!

**Review by HaraldPanten — APPROVED** [review]
I couldn't test on runbot, even rebuilding. I tested this module in local environment and I think that it's working fine 👍

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
You haven't removed `models/account_invoice.py` file.

You are also changing indentation of XML files in the migration commit, which makes the migration less readable. That's maybe due to the new XML prettier, but would be good that you come back to the commit previous to the migration, apply black + prettier, and commit on that one `[IMP] black, isort`. Backup before doing that what you have currently, and after applying pre-commit, copy again this backup and commit it as migration.

File: account_banking_mandate/README.rst (hunk 1 above)
Comment (misern2) [reply account_banking_mandate/README.rst:102]: Done

File: account_banking_mandate/migrations/12.0.2.0.0/post-migration.py (hunk 2 above)
Comment (misern2) [reply account_banking_mandate/migrations/12.0.2.0.0/post-migration.py]: Done

File: account_banking_mandate/migrations/12.0.2.0.0/pre-migration.py (hunk 3 above)
Comment (misern2) [reply account_banking_mandate/migrations/12.0.2.0.0/pre-migration.py]: Done

File: account_banking_mandate/models/account_banking_mandate.py (hunk 4 above)
Comment (misern2) [reply account_banking_mandate/models/account_banking_mandate.py]: Done

File: account_banking_mandate/models/account_banking_mandate.py (hunk 5 above)
Comment (misern2) [reply account_banking_mandate/models/account_banking_mandate.py]: Done

**Comment by misern2** [comment]
@sergio-teruel @pedrobaeza all suggested changes are applied. Thanks!

File: account_banking_mandate/models/account_move.py [hunk 6]
```diff
@@ -48,28 +39,9 @@ def create(self, vals):
                 for field in changed_fields:
                     if field not in vals and invoice[field]:
                         vals[field] = invoice._fields[field].convert_to_write(
-                            invoice[field], invoice,
+                            invoice[field], invoice
                         )
-        return super(AccountInvoice, self).create(vals)
-
-    # If a customer pays via direct debit, it's refunds should
```
Comment (pedrobaeza) [inline account_banking_mandate/models/account_move.py]: Is this code restored anywhere?

Comment (pedrobaeza) [reply account_banking_mandate/models/account_move.py]: OK, seen in the tests that is working

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
There are things that can be improved:

- Create multi for mandates
- Some onchange methods to be changed to computed with readonly

but for not blocking this more, let's merge it and that can come as later improvement:

/ocabot merge

Thanks for the migration

**Pull request merged**